### PR TITLE
KAFKA-13689: optimize the log output of logUnused method

### PR DESCRIPTION
**Pull request by RivenSun2**
optimize the log output of logUnused method.

### CODE REVIEW

**Comment by RivenSun2** [comment]
Hi @guozhangwang 
please help to review PR when available.
Thanks.

**Comment by guozhangwang** [comment]
Hi @RivenSun2 I think we also want to call `ignore` when certain producer feature is disabled as well?

**Comment by RivenSun2** [comment]
@guozhangwang 
Of course, I briefly sort out where KafkaClient calls `AbstractConfig.getBoolean` method.
Only the `KafkaProducer.configureTransactionState` method needs to be processed.
`transactional.id` and `retry.backoff.ms` are retrieved elsewhere.

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 1]
```diff
@@ -530,6 +530,9 @@ private TransactionManager configureTransactionState(ProducerConfig config,
                 log.info("Instantiated a transactional producer.");
             else
                 log.info("Instantiated an idempotent producer.");
+        } else {
+            // ignore unretrieved configurations related to producer transaction
+            config.ignore(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG);
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:535]: Should we also ignore `TRANSACTIONAL_ID_CONFIG` as well?

Comment (RivenSun2) [reply clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:535]: It is called at the beginning of the initialization method of KafkaProducer: 
 `String transactionalId = config.getString(ProducerConfig.TRANSACTIONAL_ID_CONFIG);`

do we need to ignore transactional.id here?

Comment (guozhangwang) [reply clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:535]: Ah I see, all good then.

**Review by guozhangwang — APPROVED** [review]

**Pull request merged**